src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -291,11 +293,19 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                         _controller = _controllerFactory.CreateController(controllerContext);
 
                         _arguments = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
-                        var task = _controllerArgumentBinder.BindArgumentsAsync(controllerContext, _controller, _arguments);
-                        if (task.Status != TaskStatus.RanToCompletion)
+
+                        // Perf: Avoid allocating async state machines where possible. We only need the state
+                        // machine if you need to bind properties or parameters.
+                        var actionDescriptor = _controllerContext.ActionDescriptor;
+                        if (actionDescriptor.BoundProperties.Count != 0 ||
```

Comment:
Keep the existing code that returned a completed task if there's nothing to bind?